### PR TITLE
Add workspace create workspace delete resource to terraform provider

### PR DESCRIPTION
**Pull request by ismirlia**
This pull request is to add workspace create and delete functionality to terraform to support combined services.

```
2023-11-20T17:42:39.505-0600 [DEBUG] sdk.helper_resource: Finished TestCase: test_step_number=1 test_terraform_path=/opt/homebrew/bin/terraform test_working_directory=/var/folders/8x/n_kmqfgx4k1216np6g31wb7h0000gn/T/plugintest2987059652 test_name=TestAccIBMPIWorkspaceBasic
--- PASS: TestAccIBMPIWorkspaceBasic (405.98s)
PASS
2023-11-20T17:42:39-06:00 debug layer=debugger detaching
Process 29956 has exited with status 0
Detaching
```

### CODE REVIEW

**Comment by michaelkad** [comment]
@hkantare @yussufsh  Could you help with reviewing this PR please?

File: website/docs/r/pi_workspace.html.markdown [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+- Please find [supported Regions](https://cloud.ibm.com/apidocs/power-cloud#endpoint) for endpoints.
+- If a Power cloud instance is provisioned at `lon04`, The provider level attributes should be as follows:
+  - `region` - `lon`
+  - `zone` - `lon04`
+
+## Argument reference
+
+Review the argument references that you can specify for your resource.
+
+- `id` - (String) Workspace ID.
```
Comment (yussufsh) [inline website/docs/r/pi_workspace.html.markdown:40]: `id` should go in `Attribute reference` section.

**Review by yussufsh — CHANGES_REQUESTED** [review]

File: ibm/service/power/resource_ibm_pi_workspace.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			PIWorkspaceName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
```
[the remark this answers is not in the document]
Comment (ismirlia) [reply ibm/service/power/resource_ibm_pi_workspace.go:32]: Please mark it as a future enhancement. I am currently out and will be out until next Tuesday.

**Review by yussufsh — APPROVED** [review]
lgtm
@hkantare please help with this one as well.

**Comment by hkantare** [comment]
Can you resolve conflits

**Comment by michaelkad** [comment]
> Can you resolve conflits

@hkantare resolved.

**Pull request merged**